packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -147,6 +176,9 @@ Future<void> _runCmakeGeneration({
         buildDir.path,
         '-G',
         generator,
+        '-A',
+        getCmakeWindowsArch(targetPlatform),
```

Comment:
This gives the following error if you've ran or built your Windows app before this change:

```
CMake Error: Error: generator platform: x64
Does not match the platform used previously:
Either remove the CMakeCache.txt file and CMakeFiles directory or choose a different binary directory.
Exception: Unable to generate build files
```

Users will need to run `flutter clean` to work around this. This will affect all Windows developers once when they upgrade their Flutter SDK. This will also affect Windows developers each time they provide a different `--target-platform` value. This seems too impactful.

Is the `-A` option strictly necessary? What is it used for? Can we work around this somehow?

Reply:
Yes you're right, the actual build layout for windows does not reflect architecture, so that would need to be changed.
Another solution is to clean the folder when a change in target-platform is detected, so that would be less impactful.

Maybe that can wait the second round where we implement --target-platform?
For now, people would be stuck with arm64 or x64 build after flutter bootstrapped itself.

the -A is needed as it's the one who dictates if you create an arm64 or x64 objects, that will be linked with flutter.dll in the end.